// concurrently/dist/src/defaults.js
"use strict";
// This file is meant to be a shared place for default configs.
// It's read by the flow controllers, the executable, etc.
// Refer to tests for the meaning of the different possible values.
Object.defineProperty(exports, "__esModule", { value: true });
exports.passthroughArguments = exports.timings = exports.cwd = exports.timestampFormat = exports.success = exports.restartDelay = exports.restartTries = exports.raw = exports.prefixLength = exports.prefixColors = exports.prefix = exports.nameSeparator = exports.hide = exports.maxProcesses = exports.handleInput = exports.defaultInputTarget = void 0;
exports.defaultInputTarget = 0;

exports.handleInput = false;

exports.maxProcesses = 0;

exports.hide = '';

exports.nameSeparator = ',';

exports.prefix = '';

exports.prefixColors = 'reset';

exports.prefixLength = 10;
exports.raw = false;

exports.restartTries = 0;

exports.restartDelay = 0;

exports.success = 'all';

exports.timestampFormat = 'yyyy-MM-dd HH:mm:ss.SSS';

exports.cwd = undefined;

exports.timings = false;

exports.passthroughArguments = false;
