// date-fns/constants/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.secondsInYear = exports.secondsInWeek = exports.secondsInQuarter = exports.secondsInMonth = exports.secondsInMinute = exports.secondsInHour = exports.secondsInDay = exports.quartersInYear = exports.monthsInYear = exports.monthsInQuarter = exports.minutesInHour = exports.minTime = exports.millisecondsInSecond = exports.millisecondsInMinute = exports.millisecondsInHour = exports.maxTime = exports.daysInYear = exports.daysInWeek = void 0;

var daysInWeek = 7;


exports.daysInWeek = daysInWeek;
var daysInYear = 365.2425;


exports.daysInYear = daysInYear;
var maxTime = Math.pow(10, 8) * 24 * 60 * 60 * 1000;


exports.maxTime = maxTime;
var millisecondsInMinute = 60000;


exports.millisecondsInMinute = millisecondsInMinute;
var millisecondsInHour = 3600000;


exports.millisecondsInHour = millisecondsInHour;
var millisecondsInSecond = 1000;


exports.millisecondsInSecond = millisecondsInSecond;
var minTime = -maxTime;


exports.minTime = minTime;
var minutesInHour = 60;


exports.minutesInHour = minutesInHour;
var monthsInQuarter = 3;


exports.monthsInQuarter = monthsInQuarter;
var monthsInYear = 12;


exports.monthsInYear = monthsInYear;
var quartersInYear = 4;


exports.quartersInYear = quartersInYear;
var secondsInHour = 3600;


exports.secondsInHour = secondsInHour;
var secondsInMinute = 60;


exports.secondsInMinute = secondsInMinute;
var secondsInDay = secondsInHour * 24;


exports.secondsInDay = secondsInDay;
var secondsInWeek = secondsInDay * 7;


exports.secondsInWeek = secondsInWeek;
var secondsInYear = secondsInDay * daysInYear;


exports.secondsInYear = secondsInYear;
var secondsInMonth = secondsInYear / 12;


exports.secondsInMonth = secondsInMonth;
var secondsInQuarter = secondsInMonth * 3;
exports.secondsInQuarter = secondsInQuarter;
